### DISCUSSION TITLE
Prompt Engineering for CogVLM

### DISCUSSION THREAD

**Discussion by aldoz-mila** [THUDM/CogVLM#312] [Q&A] [answered]
Hello, I was wondering if there are any guidelines, or experiences from users, on the best prompt formats to use when doing inference with CogVLM? Some other models have for example an expected format in the text prompt in chat/conversation settings. I am trying to optimize my current prompts and am open to any ideas.

**Comment by zRzRzRzRzRzRzR** [comment, marked as answer]
We also have our formats, which you can see [here](https://github.com/THUDM/CogVLM/blob/main/composite_demo/utils.py)

**Reply by aldoz-mila** [reply]
Thank you, this was exactly what I was looking for! :)
